pca_test.go

```diff
@@ -0,0 +1,282 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stat
+
+import (
+	"testing"
+
+	"github.com/gonum/floats"
+	"github.com/gonum/matrix/mat64"
+)
+
+func TestPrincipalComponents(t *testing.T) {
+	for i, test := range []struct {
+		data     mat64.Matrix
+		k        int
+		wantPC   *mat64.Dense
+		wantProj *mat64.Dense
+		wantVars []float64
+		epsilon  float64
+	}{
+		// Test results manually verified using octave.
```

Comment:
If you could provide a case (also a weighted test case would be helpful), I'll add it.